common/types.go

```diff
@@ -117,6 +119,7 @@ func (h Hash) Generate(rand *rand.Rand, size int) reflect.Value {
 	return reflect.ValueOf(h)
 }
 
+// EmptyHash checks whether h is the zero hash.
 func EmptyHash(h Hash) bool {
 	return h == Hash{}
 }
```

Comment:
Please delete this method, there's really no point to have this method (also it's a pointless function call overhead).